--- a/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi
@@ -7,6 +7,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/pinctrl/rzg2l-pinctrl.h>
+#include <dt-bindings/net/mscc-phy-vsc8531.h>
 
 /* SW1[2] should be at position 2/OFF to enable 64 GB eMMC */
 #define EMMC	1
@@ -50,6 +51,7 @@
 / {
 	aliases {
 		i2c3 = &i2c3;
+		i2c2 = &i2c2;
 		ethernet0 = &eth0;
 		ethernet1 = &eth1;
 	};
@@ -151,26 +153,20 @@
 	phy-mode = "rgmii-id";
 	status = "okay";
 
-	phy0: ethernet-phy@7 {
-		compatible = "ethernet-phy-id0022.1640",
-			     "ethernet-phy-ieee802.3-c22";
-		reg = <7>;
-		rxc-skew-psec = <2400>;
-		txc-skew-psec = <2400>;
-		rxdv-skew-psec = <0>;
-		txdv-skew-psec = <0>;
-		rxd0-skew-psec = <0>;
-		rxd1-skew-psec = <0>;
-		rxd2-skew-psec = <0>;
-		rxd3-skew-psec = <0>;
-		txd0-skew-psec = <0>;
-		txd1-skew-psec = <0>;
-		txd2-skew-psec = <0>;
-		txd3-skew-psec = <0>;
-
-		interrupt-parent = <&pinctrl>;
-		interrupts = <RZG2L_GPIO(1, 0) IRQ_TYPE_LEVEL_LOW>;
-	};
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+ 
+ 		interrupt-parent = <&pinctrl>;
+		interrupts = <RZG2L_GPIO(3, 0) IRQ_TYPE_EDGE_FALLING>;
+	};
+	
+	vsc8531_0: ethernet-phy@0 {
+                 compatible = "ethernet-phy-id0007.0570";
+                 vsc8531,vddmac		= <3300>;
+                 vsc8531,edge-slowdown	= <7>;
+                 vsc8531,led-0-mode	= <VSC8531_LINK_ACTIVITY>;
+                 vsc8531,led-1-mode	= <VSC8531_LINK_1000_ACTIVITY>;
+        };
 };
 
 &eth1 {
@@ -180,26 +176,20 @@
 	phy-mode = "rgmii-id";
 	status = "okay";
 
-	phy1: ethernet-phy@7 {
-		compatible = "ethernet-phy-id0022.1640",
-			     "ethernet-phy-ieee802.3-c22";
-		reg = <7>;
-		rxc-skew-psec = <2400>;
-		txc-skew-psec = <2400>;
-		rxdv-skew-psec = <0>;
-		txdv-skew-psec = <0>;
-		rxd0-skew-psec = <0>;
-		rxd1-skew-psec = <0>;
-		rxd2-skew-psec = <0>;
-		rxd3-skew-psec = <0>;
-		txd0-skew-psec = <0>;
-		txd1-skew-psec = <0>;
-		txd2-skew-psec = <0>;
-		txd3-skew-psec = <0>;
+	phy1: ethernet-phy@0 {
+		reg = <0>;
 
 		interrupt-parent = <&pinctrl>;
-		interrupts = <RZG2L_GPIO(1, 1) IRQ_TYPE_LEVEL_LOW>;
+		interrupts = <RZG2L_GPIO(3, 1) IRQ_TYPE_LEVEL_LOW>;
 	};
+	
+	vsc8531_1: ethernet-phy@0 {
+                 compatible = "ethernet-phy-id0007.0572";
+                 vsc8531,vddmac		= <3300>;
+                 vsc8531,edge-slowdown	= <7>;
+                 vsc8531,led-0-mode	= <VSC8531_LINK_ACTIVITY>;
+                 vsc8531,led-1-mode	= <VSC8531_LINK_1000_ACTIVITY>;
+        };
 };
 
 &extal_clk {
@@ -261,7 +251,6 @@
 
 	gpio-sd0-pwr-en-hog {
 		gpio-hog;
-		gpios = <RZG2L_GPIO(4, 1) GPIO_ACTIVE_HIGH>;
 		output-high;
 		line-name = "gpio_sd0_pwr_en";
 	};
@@ -287,7 +276,7 @@
 	 */
 	sd0-dev-sel-hog {
 		gpio-hog;
-		gpios = <RZG2L_GPIO(41, 1) GPIO_ACTIVE_HIGH>;
+		gpios = <RZG2L_GPIO(8, 1) GPIO_ACTIVE_HIGH>;
 		output-high;
 		line-name = "sd0_dev_sel";
 	};
@@ -479,8 +468,8 @@
 	memory-region = <&global_cma>;
 };
 
-&i2c3 {
-	pinctrl-0 = <&i2c3_pins>;
+&i2c2 {
+	pinctrl-0 = <&i2c2_pins>;
 	pinctrl-names = "default";
 	clock-frequency = <400000>;
 
