--- a/arch/arm64/boot/dts/renesas/rz-smarc-common.dtsi
+++ b/arch/arm64/boot/dts/renesas/rz-smarc-common.dtsi
@@ -255,6 +255,24 @@
 	status = "okay";
 };
 
+&spi0 {
+	pinctrl-0 = <&spi0_pins>;
+	pinctrl-names = "default";
+	dmas = <&dmac 0x2e95>, <&dmac 0x2e96>;
+	dma-names = "tx", "rx";
+
+	status = "okay";
+
+	da16xxx@0 {
+		compatible = "renesas,da16xxx";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+		reset-gpios = <&pinctrl RZG2L_GPIO(14, 0) GPIO_ACTIVE_LOW>;
+		irq0-gpios = <&pinctrl RZG2L_GPIO(0, 0) GPIO_ACTIVE_LOW>;
+		irq1-gpios = <&pinctrl RZG2L_GPIO(1, 1) GPIO_ACTIVE_LOW>;
+	};
+};
+
 #if (!MTU3_PWM_SUPPORT)
 &spi1 {
 	pinctrl-0 = <&spi1_pins>;
